python/ffsim/protocols/trace_protocol.py

```diff
@@ -45,3 +51,72 @@ def trace(obj: Any, norb: int, nelec: tuple[int, int]) -> float:
         "The object did not have a _trace_ method that returned the trace "
         "or a _diag_ method that returned its diagonal entries."
     )
+
+
+def _trace_term(
+    op: tuple[tuple[bool, bool, int], ...], norb: int, nelec: tuple[int, int]
+):
+    n_elec_alpha, n_elec_beta = nelec
+
+    combined_indices = set([(orb, spin) for _, spin, orb in op])
+    norb_alpha = sum([not spin for _, spin in combined_indices])
+    norb_beta = sum([spin for _, spin in combined_indices])
```

Comment:
```suggestion
    norb_beta = len(combined_indices) - norb_alpha
```
to avoid iterating through `combined_indices` again.